go-controller/pkg/node/management-port_linux.go

```diff
@@ -128,3 +130,95 @@ func DelMgtPortIptRules() {
 	_ = ipt.DeleteChain("nat", iptableMgmPortChain)
 	_ = ipt6.DeleteChain("nat", iptableMgmPortChain)
 }
+
+// checks to make sure that following configurations are present on the k8s node
+// 1. route entries to cluster CIDR and service CIDR through management port
+// 2. ARP entry for the node subnet's gateway ip
+// 3. IPtables chain and rule for SNATing packets entering the logical topology
+func checkManagementPortHealth(portIP, routerIP, routerMac string, stopChan chan struct{}) {
+	var link netlink.Link
+	var ipt util.IPTablesHelper
+	var err error
+
+	subnets := []string{config.Kubernetes.ServiceCIDR}
+	for _, subnet := range config.Default.ClusterSubnets {
+		subnets = append(subnets, subnet.CIDR.String())
+	}
+	ticker := time.NewTicker(30 * time.Second)
+	for {
+		select {
+		case <-ticker.C:
+			if link == nil {
+				if link, err = netlink.LinkByName(util.K8sMgmtIntfName); err != nil {
+					klog.Errorf("failed to lookup link %s: %v", util.K8sMgmtIntfName, err)
+					continue
+				}
+			}
+
+			if exists, err := util.LinkNeighExists(link, routerIP, routerMac); err == nil {
```

Comment:
@girishmg instead of doing effectively the same code over again, could we combine this part with the existing setup code and just have one function? We only do the management port setup once so it's not really performance sensitive; we could just run the "does the route exist" check before we add the route like this health check does.

I'd rather have one piece of code that's longer/a bit more complex than two pieces that do mostly the same thing...

Reply:
done